tests/framework/artifacts.py

```diff
@@ -1,220 +1,87 @@
 # Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
-"""Define classes for interacting with CI artifacts in s3."""
 
-import functools
-import os
-import platform
-import tempfile
+"""Define classes for interacting with CI artifacts"""
+
+import re
 from dataclasses import dataclass
-from enum import Enum
 from pathlib import Path
-from shutil import copyfile
-from stat import S_IREAD, S_IWRITE
+from typing import Iterator
 
-import boto3
-import botocore.client
+import pytest
 
 import host_tools.network as net_tools
-from framework.defs import (
-    DEFAULT_TEST_SESSION_ROOT_PATH,
-    SUPPORTED_KERNELS,
-    SUPPORTED_KERNELS_NO_SVE,
-)
-from framework.utils import compare_versions, get_kernel_version
-from framework.utils_cpuid import get_instance_type
-from host_tools.cargo_build import run_rebase_snap_bin
-
-ARTIFACTS_LOCAL_ROOT = f"{DEFAULT_TEST_SESSION_ROOT_PATH}/ci-artifacts"
+from framework.defs import ARTIFACT_DIR
+from framework.properties import global_props
+from framework.utils import get_firecracker_version_from_toml
 
 
 def select_supported_kernels():
```

Comment:
Mh, the "SUPPORTED_KERNELS(.*)" constants in `defs.py` seem to just be memoization of this function. Could this just be "if c7g and 4.14 return SUPPORTED_NO_SVE else return SUPPORTED_YES_SVE"?

Reply:
Actually those constants are no longer used... will remove